che-terminal-client/pom.xml

```diff
@@ -57,42 +57,45 @@
                 </configuration>
             </plugin>
             <plugin>
-                <artifactId>maven-antrun-plugin</artifactId>
+                <groupId>com.github.eirslett</groupId>
+                <artifactId>frontend-maven-plugin</artifactId>
                 <executions>
                     <execution>
-                        <!--clean-up build folders-->
-                        <id>clean-up</id>
-                        <phase>clean</phase>
+                        <id>install node and npm</id>
+                        <phase>generate-resources</phase>
                         <goals>
-                            <goal>run</goal>
+                            <goal>install-node-and-npm</goal>
                         </goals>
                         <configuration>
-                            <tasks>
-                                <delete dir="${basedir}/node_modules" />
-                                <delete dir="${basedir}/build" />
-                            </tasks>
+                            <arguments>install --fetch-retry-mintimeout 60000 --fetch-retries 10 --maxsockets 2</arguments>
                         </configuration>
                     </execution>
                     <execution>
-                        <!--compilation sources-->
-                        <id>compilation</id>
+                        <id>npm install</id>
                         <phase>compile</phase>
                         <goals>
-                            <goal>run</goal>
+                            <goal>npm</goal>
                         </goals>
                         <configuration>
-                            <target>
-                                Download NPM dependencies
-                                <exec dir="${basedir}" executable="npm" failonerror="true">
-                                    <arg value="install" />
-                                </exec>
-                                <exec dir="${basedir}" executable="npm" failonerror="true">
-                                    <arg value="run-script" />
-                                    <arg value="build" />
-                                </exec>
-                            </target>
+                            <arguments>install --save-dev --fetch-retry-mintimeout 60000 --fetch-retries 10 --maxsockets 2</arguments>
+                            <npmRegistryURL>http://registry.npmjs.org/</npmRegistryURL>
```

Comment:
why is it http without https ?

Reply:
because NCL doesn't support https, so we need this override to allow it to work when building in NCL. See https://docs.engineering.redhat.com/display/JP/Running+NPM+and+Yarn+builds+with+maven+frontend+plugin for explanations.